pandas/tseries/period.py

```diff
@@ -211,11 +210,46 @@ def __new__(cls, data=None, ordinal=None, freq=None, start=None, end=None,
             else:
                 data, freq = cls._generate_range(start, end, periods,
                                                  freq, kwargs)
-        else:
```

Comment:
can you add under ``_simple_new`` and ``_from_ordinals`` pointers to each other and an expl of when to use which for construction (or can do in a block above them both).